### DISCUSSION TITLE
How to link to the next post within a folder

### DISCUSSION THREAD

**Discussion by rhyses-pieces** [11ty/eleventy#2703] [Q&A] [answered]
Hiya! I'm digging 11ty but I'm running into some trouble.

I have a site that's currently set up with folders within folders. This is what it looks like roughly:

```
src/
├─ posts/
├─ writing/
│  ├─ book-title/
│  │  ├─ chapter-1.md
│  │  ├─ chapter-2.md
│  ├─ book-title-2/
│  │  ├─ chapter-1.md
│  │  ├─ chapter-2.md
```

I want to be able to automatically link to the next chapter within my layout for each chapter page instead of having to manually link each chapter. How can I achieve this? I've looked everywhere and got confused by a lot of different approaches.

EDIT: I'd also like to make a list of links of each book, like a table of contents, and have that also available on each chapter page but that's another headache for another day.

**Comment by pdehaan** [comment, marked as answer]
What about making each book its own collection, and then using something like https://www.11ty.dev/docs/filters/collection-items/#get-next-or-previous-collection-item-universal-filters for chapter navigation.

**Reply by pdehaan** [reply]
It's a bit messy, but this might give you some ideas (either good ideas or bad ideas). 🤷 
https://github.com/pdehaan/11ty-2703

**Reply by rhyses-pieces** [reply]
I've tried a method similar to yours, but I end up with a result that doesn't generate my actual book. I don't know if it's because I have it set to a folder deeper in the writing folder (`./src/writing/books/`) but it seems to work if I have all the books set to just within the writing folder.

EDIT: after some tweaking to fit my setup, it only generates the collections for two other folders, nothing within the `writing` folder. I'll upload what I have later tonight.

EDIT2: [here's my repo](https://github.com/rhyses-pieces/11ty)

**Reply by pdehaan** [reply]
I had to modify the repo a bit to get it to install and build, but I wasn't seeing the /src/writing/books/one/** files building because of some pagination in the layout front matter:

```njk
---
# pagination:
#   data: collections.fic
#   size: 2
---
```

**Reply by rhyses-pieces** [reply]
doh - thanks for the catch. I deleted that and it seems to work now - thanks a lot for the insight!!

**Comment by pdehaan** [comment]
As for the table of contents (toc) idea, since you're using .md files, you could look at something like https://www.npmjs.com/package/markdown-it-toc (or one of the other probably dozen) TOC plugins, but that would probably only work on an individual file (so a chapter based TOC, which might not be entirely helpful).
